plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -515,6 +504,24 @@ func ClusterRoles() []rbacv1.ClusterRole {
 		},
 	}...)
 
+	kubeletAPIAdminRules := []rbacv1.PolicyRule{
+		// Allow read-only access to the Node API objects
+		rbacv1helpers.NewRule("get", "list", "watch").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
+		// Allow all API calls to the nodes
+		rbacv1helpers.NewRule("proxy").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
+		rbacv1helpers.NewRule("*").Groups(legacyGroup).Resources("nodes/proxy", "nodes/metrics", "nodes/stats", "nodes/log").RuleOrDie(),
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ContainerCheckpoint) {
```

Comment:
Can you add a unit test covering the feature gate case? You should verify that when feature gate is set we see this checkpoint. And feature gate off we should not.